include/extractor/guidance/intersection.hpp

```diff
@@ -59,7 +60,73 @@ struct ConnectedRoad final
 // small helper function to print the content of a connected road
 std::string toString(const ConnectedRoad &road);
 
-typedef std::vector<ConnectedRoad> Intersection;
+struct Intersection
```

Comment:
Nice - thank you for starting a refactor here! You could probably make your life easier by doing

```cpp
struct Intersection final : vector<Road> {
  using Base = vector<Road>;

  using Base::size;
  ..
};
```